src/Microsoft.ML.Data/DataLoadSave/Binary/BinaryLoaderSaverCatalog.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+
+namespace Microsoft.ML
+{
+    public static class BinaryLoaderSaverCatalog
+    {
+        /// <summary>
+        /// Read a data view from a binary file using <see cref="BinaryLoader"/>.
+        /// </summary>
+        /// <param name="catalog">The catalog.</param>
+        /// <param name="args">The arguments to binary reader.</param>
+        /// <param name="path"></param>
+        public static IDataView ReadFromBinaryFile(this DataOperations catalog,
+            BinaryLoader.Arguments args, string path)
```

Comment:
>path [](start = 48, length = 4)

I think we should fix both this and `ReadFromText` to read from a stream.

Reply:
@Zruty0 I've changed this to read from a `Stream`. For `ReadFromTextFile`, I see that `TextLoader.Read` accepts `IMultiStreamSource` (of which `MultiFileSource` seems to be the only implementation that shows up in the codebase) and not `Stream`. Should I change `ReadFromTextFile` to read from a `MultiFileSource` instead?

Reply:
For these extensions, use `Stream` (and not stream sources). Also, call them `ReadFromBla` where `Bla` is `Binary` or `Text`

---
In reply to: [235158400](https://github.com/dotnet/machinelearning/pull/1678#discussion_r235158400) [](ancestors = 235158400)

Reply:
There is no `TextLoader.Read` method that accepts a `Stream`. Do you suggest that we make `ReadFromText` accept a `FileStream`, then get the path to the file with `FileStream.Name`, and return 
```csharp 
reader.Read(new MultiFileSource(path)) 
```
?

Reply:
Oh, right, I got it now. Ok, let's not touch `Text` then, and let's have `Binary` work from both streams and files.